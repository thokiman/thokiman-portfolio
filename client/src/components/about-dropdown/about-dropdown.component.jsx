import React from "react";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import {
  selectAboutRoute,
  selectTimelineCareerRoute,
} from "../../redux/about/about.selectors";
import AboutDropdownBar from "../about-dropdown-content/about-dropdown-bar.component";
import { AboutDropdownContainer } from "./about-dropdown.styles";
import { selectIsSideBarHidden } from "../../redux/header/header.selectors";
export const AboutDropdown = ({
  location: { pathname },
  aboutRoute,
  timelineCareerRoute,
  isSideBarHidden,
}) => {
  return (
    <AboutDropdownContainer
      $matchpath={!!pathname.match(`${aboutRoute}${timelineCareerRoute}`)}
      $issidebarhidden={isSideBarHidden}
    >
      <AboutDropdownBar />
    </AboutDropdownContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  timelineCareerRoute: selectTimelineCareerRoute,
  aboutRoute: selectAboutRoute,
  isSideBarHidden: selectIsSideBarHidden,
});
export default withRouter(connect(mapStateToProps)(AboutDropdown));
