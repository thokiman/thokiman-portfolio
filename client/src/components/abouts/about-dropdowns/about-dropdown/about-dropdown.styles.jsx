import styled, { css } from 'styled-components';

const getTimelineImage = ({ $lowWaveImage }) => {
  return $lowWaveImage.item.iconPath;
};

const SideBarActive = css`
  display: none;
`;
const getSideBarActive = (props) => {
  if (props.$issidebarhidden) return SideBarActive;
  return null;
};

const AboutDropdownStyles = css`
  width: 30vw;
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  margin-left: 2vw;
  transition: height 0.5s ease-in-out, top 0.2s ease;
  @media only screen and (max-width: 800px) {
    width: 100%;
    height: 50vh;
    margin-left: 0;
    margin-top: 20px;
    ${getSideBarActive}
  }
`;

const AboutDropdownLeftPageContainer = css`
  ${AboutDropdownStyles}

  height: 98%;
`;

const AboutDropdownLeftPageContainerTimeline = css`
  ${AboutDropdownStyles}
  height: 580px;
  width: 320px;
  background: url(${getTimelineImage});
  background-size: cover;
  @media only screen and (max-width: 800px) {
    background-image: none;
  }
`;

const getAboutDropdownContainer = ({ $matchpath }) => {
  if ($matchpath) {
    return AboutDropdownLeftPageContainerTimeline;
  }
  return AboutDropdownLeftPageContainer;
};

export const AboutDropdownContainer = styled.div`
  ${getAboutDropdownContainer}
`;
AboutDropdownContainer.displayName = 'AboutDropdownContainer';
