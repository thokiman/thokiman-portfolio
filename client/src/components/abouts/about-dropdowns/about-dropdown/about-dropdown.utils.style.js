import {
  hasScreenSize1,
  hasScreenSize2,
  hasScreenSize3_1,
  hasScreenSize3_2,
  hasScreenSize3_3,
  hasScreenSize3_4,
  hasScreenSize3_5,
  hasScreenSize4,
  hasScreenSize5,
} from '../../../../utils/common.component.utils';

export const measureAboutDropdownContainerProps = (viewWidth, viewHeight) => {
  const aboutDropdownContainerHeightOffset = 0;
  const aboutDropdownContainerTopOffset = 0;
  let abouDropdownContainerProps;

  if (hasScreenSize1(viewWidth)) {
    console.log('about-dropdown', 1);
    abouDropdownContainerProps = {
      height: 52 + aboutDropdownContainerHeightOffset,
      top: 0 + aboutDropdownContainerTopOffset,
    };
  } else if (hasScreenSize2(viewWidth)) {
    console.log('about-dropdown', 2);
    abouDropdownContainerProps = {
      height: 49 + aboutDropdownContainerHeightOffset,
      top: 0 + aboutDropdownContainerTopOffset,
    };
  } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
    console.log('about-dropdown', 3);
    abouDropdownContainerProps = {
      height: 42 + aboutDropdownContainerHeightOffset,
      top: 0 + aboutDropdownContainerTopOffset,
    };
  } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
    console.log('about-dropdown', 4);
    abouDropdownContainerProps = {
      height: 52 + aboutDropdownContainerHeightOffset,
      top: 0 + aboutDropdownContainerTopOffset,
    };
  } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
    console.log('about-dropdown', 5);
    abouDropdownContainerProps = {
      height: 42 + aboutDropdownContainerHeightOffset,
      top: 0 + aboutDropdownContainerTopOffset,
    };
  } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
    console.log('about-dropdown', 6);
    abouDropdownContainerProps = {
      height: 42 + aboutDropdownContainerHeightOffset,
      top: 0 + aboutDropdownContainerTopOffset,
    };
  } else if (hasScreenSize3_5(viewWidth, viewHeight)) {
    console.log('about-dropdown', 7);
    abouDropdownContainerProps = {
      height: 42 + aboutDropdownContainerHeightOffset,
      top: 0 + aboutDropdownContainerTopOffset,
    };
  } else if (hasScreenSize4(viewWidth, viewHeight)) {
    console.log('about-dropdown', 8);
    abouDropdownContainerProps = {
      height: 42 + aboutDropdownContainerHeightOffset,
      top: 0 + aboutDropdownContainerTopOffset,
    };
  } else if (hasScreenSize5(viewWidth)) {
    console.log('about-dropdown', 9);
    abouDropdownContainerProps = {
      height: 49 + aboutDropdownContainerHeightOffset,
      top: 0 + aboutDropdownContainerTopOffset,
    };
  } else {
    console.log('about-dropdown', 10);
    abouDropdownContainerProps = {
      height: 60,
      top: 1,
    };
  }

  return abouDropdownContainerProps;
};
