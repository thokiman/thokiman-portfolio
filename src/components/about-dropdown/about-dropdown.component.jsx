import React from "react";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import {
  selectAboutRoute,
  selectTimelineCareerRoute,
} from "../../redux/about/about.selectors";
import AboutDropdownBar from "../about-dropdown-content/about-dropdown-bar.component";
import "./about-dropdown.styles.scss";

const AboutDropdown = ({
  location: { pathname },
  timelineCareerRoute,
  aboutRoute,
}) => {
  return (
    <div
      className={
        pathname.match(`${aboutRoute}${timelineCareerRoute}`)
          ? "left-page-container-timeline"
          : "left-page-container"
      }
    >
      <AboutDropdownBar />
    </div>
  );
};

const mapStateToProps = createStructuredSelector({
  timelineCareerRoute: selectTimelineCareerRoute,
  aboutRoute: selectAboutRoute,
});
export default withRouter(connect(mapStateToProps)(AboutDropdown));
