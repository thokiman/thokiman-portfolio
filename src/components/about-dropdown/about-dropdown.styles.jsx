import styled, { css } from "styled-components";
import TimelineImage from "../../assets/layout/water-2.jpg";
const AboutDropdownStyles = css`
  width: 30vw;
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  margin-left: 2vw;
  transition: height 0.5s ease-in-out, top 0.2s ease;
`;

const AboutDropdownLeftPageContainer = css`
  ${AboutDropdownStyles}
  height: 98%;
`;

const AboutDropdownLeftPageContainerTimeline = css`
  ${AboutDropdownStyles}
  height: 580px;
  width: 320px;
  background: url(${TimelineImage});
  background-size: cover;
`;

const getAboutDropdownContainer = ({ $matchpath }) => {
  if ($matchpath) {
    return AboutDropdownLeftPageContainerTimeline;
  }
  return AboutDropdownLeftPageContainer;
};

export const AboutDropdownContainer = styled.div`
  ${getAboutDropdownContainer}
`;
AboutDropdownContainer.displayName = "AboutDropdownContainer";
