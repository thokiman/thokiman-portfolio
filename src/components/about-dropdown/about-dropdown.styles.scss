.left-page-container {
    width: 25%;
    height: 98%;
    display: flex;
    align-items: center;
    justify-content: center;
    text-align: center;
    background-color: rgba($color: #128194, $alpha: .5);
    position: relative;
    left: 40px;
    transition: height 0.5s ease-in-out, top 0.2s ease;


}

.left-page-container-timeline {
    width: 25%;
    height: 14%;
    display: flex;
    align-items: center;
    justify-content: center;
    text-align: center;
    background-color: rgba($color: #128194, $alpha: .5);
    position: relative;
    top: -1785px;
    left: 40px;
    transition: height 0.5s ease-in-out, top 0.2s ease;

}