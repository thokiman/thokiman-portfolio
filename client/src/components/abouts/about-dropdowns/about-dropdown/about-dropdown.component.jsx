import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { createStructuredSelector } from 'reselect';
import {
  selectAboutRoute,
  selectLowWaveImage,
  selectTimelineCareerRoute,
} from '../../../../redux/about/about.selectors';
import { selectIsSideBarActive } from '../../../../redux/header/header.selectors';
import AboutDropdownBar from '../about-dropdown-content/about-dropdown-bar.component';
import { AboutDropdownContainer } from './about-dropdown.styles';

export const AboutDropdown = ({
  location: { pathname },
  aboutRoute,
  timelineCareerRoute,
  isSideBarActive,
  lowWaveImage,
}) => {
  return (
    <AboutDropdownContainer
      $matchpath={!!pathname.match(`${aboutRoute}${timelineCareerRoute}`)}
      $issidebaractive={isSideBarActive}
      $lowWaveImage={lowWaveImage}
    >
      <AboutDropdownBar />
    </AboutDropdownContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  timelineCareerRoute: selectTimelineCareerRoute,
  aboutRoute: selectAboutRoute,
  isSideBarActive: selectIsSideBarActive,
  lowWaveImage: selectLowWaveImage,
});
export default withRouter(connect(mapStateToProps)(AboutDropdown));
