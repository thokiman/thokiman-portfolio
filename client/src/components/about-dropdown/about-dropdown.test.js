import React from "react";
import { shallow } from "enzyme";

import { AboutDropdown } from "./about-dropdown.component";

describe("AboutDropdown component", () => {
  let wrapper;
  const mockText = "";
  const mockFalseBool = false;
  beforeEach(() => {
    const mockProps = {
      location: { pathname: mockText },
      aboutRoute: mockText,
      timelineCareerRoute: mockText,
      isSideBarHidden: mockFalseBool,
    };
    wrapper = shallow(<AboutDropdown {...mockProps} />);
  });

  it("should render AboutDropdown component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
