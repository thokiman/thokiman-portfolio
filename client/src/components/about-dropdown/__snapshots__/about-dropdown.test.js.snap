// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutDropdown component should render AboutDropdown component 1`] = `ShallowWrapper {}`;
